### PR TITLE
lock sessions executors

### PR DESCRIPTION
**Pull request by itayd**
replaces #200

### CODE REVIEW

File: internal/backend/sessions/sessioncalls/calls.go [hunk 1]
```diff
@@ -58,6 +55,11 @@ type calls struct {
 	//                 user script.
 	generalWorker worker.Worker
 	uniqueWorker  worker.Worker
+
+	// Pass executors for session specific calls (functions that are
+	// defined in runtime script, as opposed to integrations).
+	executorsForSessions   map[sdktypes.SessionID]*sdkexecutor.Executors
+	executorsForSessionsMu sync.RWMutex
```
Comment (tebeka) [inline internal/backend/sessions/sessioncalls/calls.go:61]: The usual practice is to place the lock variable on top of the variables it guards.

File: internal/backend/sessions/sessioncalls/calls.go [hunk 2]
```diff
@@ -76,11 +78,12 @@ func New(z *zap.Logger, config Config, svcs *sessionsvcs.Svcs) Calls {
...
-		config:        config,
-		svcs:          svcs,
-		generalWorker: worker.New(svcs.Temporal, taskQueueName, opts),
-		uniqueWorker:  worker.New(svcs.Temporal, uniqueWorkerCallTaskQueueName(), opts),
+		z:                    z,
+		config:               config,
+		svcs:                 svcs,
+		generalWorker:        worker.New(svcs.Temporal, taskQueueName, opts),
+		uniqueWorker:         worker.New(svcs.Temporal, uniqueWorkerCallTaskQueueName(), opts),
+		executorsForSessions: make(map[sdktypes.SessionID]*sdkexecutor.Executors, 16),
```
Comment (tebeka) [inline internal/backend/sessions/sessioncalls/calls.go:86]: Why 16?

**Review by tebeka — APPROVED** [review]

File: internal/backend/sessions/sessioncalls/calls.go (hunk 2 above)
Comment (itayd) [reply internal/backend/sessions/sessioncalls/calls.go:86]: no real reason. just to prep the map with some capacity.

File: internal/backend/sessions/sessioncalls/calls.go (hunk 1 above)
Comment (itayd) [reply internal/backend/sessions/sessioncalls/calls.go:61]: news to me, but ok

**Pull request merged**